libethcore/EthashAux.cpp

```diff
@@ -111,114 +84,101 @@ void EthashAux::killCache(h256 const& _s)
 }
 
-EthashAux::LightType EthashAux::light(h256 const& _seedHash)
+EthashAux::LightType EthashAux::light(uint64_t _blockNumber)
 {
 	RecursiveGuard l(get()->x_this);
-	LightType ret = get()->m_lights[_seedHash];
-	return ret ? ret : (get()->m_lights[_seedHash] = make_shared<LightAllocation>(_seedHash));
+	h256 seedHash = EthashAux::seedHash(_blockNumber);
+	LightType ret = get()->m_lights[seedHash];
+	return ret ? ret : (get()->m_lights[seedHash] = make_shared<LightAllocation>(_blockNumber));
 }
 
-EthashAux::LightAllocation::LightAllocation(h256 const& _seed)
+EthashAux::LightAllocation::LightAllocation(uint64_t _blockNumber)
 {
-	auto p = params(_seed);
-	size = p.cache_size;
-	light = ethash_new_light(&p, _seed.data());
+	light = ethash_light_new(_blockNumber);
+	size = ethash_get_cachesize(_blockNumber);
 }
 
 EthashAux::LightAllocation::~LightAllocation()
 {
-	ethash_delete_light(light);
+	ethash_light_delete(light);
 }
 
-EthashAux::FullType EthashAux::full(BlockInfo const& _header, bytesRef _dest, bool _createIfMissing)
+bytesConstRef EthashAux::LightAllocation::data() const
 {
-	return full(_header.seedHash(), _dest, _createIfMissing);
+	return bytesConstRef((byte const*)light->cache, size);
 }
 
-EthashAux::FullType EthashAux::full(h256 const& _seedHash, bytesRef _dest, bool _createIfMissing)
+EthashAux::FullAllocation::FullAllocation(ethash_light_t _light, ethash_callback_t _cb)
 {
-	RecursiveGuard l(get()->x_this);
-	FullType ret = get()->m_fulls[_seedHash].lock();
-	if (ret && _dest)
-	{
-		assert(ret->data.size() <= _dest.size());
-		ret->data.copyTo(_dest);
-		return FullType();
-	}
-	if (!ret)
-	{
-		// drop our last used cache sine we're allocating another 1GB.
-		get()->m_lastUsedFull.reset();
+	full = ethash_full_new(_light, _cb);
+}
 
-		try {
-			boost::filesystem::create_directories(getDataDir("ethash"));
-		} catch (...) {}
+EthashAux::FullAllocation::~FullAllocation()
+{
+	ethash_full_delete(full);
+}
 
-		auto info = rlpList(Ethash::revision(), _seedHash);
-		std::string oldMemoFile = getDataDir("ethash") + "/full";
-		std::string memoFile = getDataDir("ethash") + "/full-R" + toString(ETHASH_REVISION) + "-" + toHex(_seedHash.ref().cropped(0, 8));
-		if (boost::filesystem::exists(oldMemoFile) && contents(oldMemoFile + ".info") == info)
-		{
-			// memofile valid - rename.
-			boost::filesystem::rename(oldMemoFile, memoFile);
-		}
+bytesConstRef EthashAux::FullAllocation::data() const
+{
+	return bytesConstRef((byte const*)ethash_full_dag(full), size());
+}
 
-		DEV_IGNORE_EXCEPTIONS(boost::filesystem::remove(oldMemoFile));
-		DEV_IGNORE_EXCEPTIONS(boost::filesystem::remove(oldMemoFile + ".info"));
+EthashAux::FullType EthashAux::full(BlockInfo const& _header)
+{
+	return full((uint64_t) _header.number);
+}
 
-		ethash_params p = params(_seedHash);
-		assert(!_dest || _dest.size() >= p.full_size);	// must be big enough.
+struct DAGChannel: public LogChannel { static const char* name(); static const int verbosity = 0; };
```

Comment:
verbosity = 0 meaning it always prints the log right? So there is no way to disable it?
On my long time running (no maintenance) Raspberry Pi I don't want any logs.